frontend/src/lib/components/BranchButton.svelte

```diff
@@ -0,0 +1,431 @@
+			await fetchExistingBranches();
+		nonProtectedBranches = fetchedNonProtected;
+		protectedBranches = fetchedProtected;
+	});
+
+	async function setBranchName(input: string) {
+		if (!input) return;
+
+		// Define validation rules
+		const maxLength = 255; // Maximum length for branch name
+		const invalidCharacters = /[~^:?*<>|]/; // Invalid special characters
+		const startsWithLetterOrNumber = /^[a-zA-Z0-9]/; // Starts with letter or number
+		const containsSpaces = /\s/; // Contains spaces
+
+		// Define an array of rules
+		const rules = [
+			'Branch names must start with a letter or a number.',
+			'Branch names cannot contain spaces. Use dashes (-) or underscores (_) instead.',
+			'Branch names cannot contain special characters like ~, ^, :, ?, *, and others.',
+			`Branch names must be shorter than ${maxLength} characters.`
+		];
+
+		const isValidBranchName = (name: string) => {
+			return (
+				!containsSpaces.test(name) &&
+				startsWithLetterOrNumber.test(name) &&
+				!invalidCharacters.test(name) &&
+				name.length <= maxLength
+			);
+		};
+
+		// Validate branch name
+		if (!isValidBranchName(input)) {
+			addToast({
+				message: 'Please ensure your branch name follows these rules:\n' + rules.join('\n'),
+				type: ToastType.Warning,
+				dismissible: true
+			});
+			return;
+		}
+
+		if (protectedBranches.some((branch) => branch.name === input)) {
+			addToast({
+				message:
+					'Please select an existing branch name from the list of non-protected branches.\n You can also create your own',
+				type: ToastType.Warning,
+				dismissible: true,
+				timeout: 1500
+			});
+			return;
+		}
+
+		// Set branch name and reset state
+		branchName.set(input);
+		newBranchName = '';
+		showMenu = false;
+
+		if (!nonProtectedBranches.some((branch) => branch.name === input)) {
+			addToast({
+				message: `Now working on a new branch: "${input}".`,
+				type: ToastType.Success,
+				dismissible: true,
+				timeout: 1800
+			});
+			return;
+		}
+
+		// Call backend to update working directory by checking out the branch
+		const response = await fetch(`${apiAddress}/api/checkout/branches/${input}`, {
+			method: 'PUT',
+			credentials: 'include'
+		});
+
+		if (!response.ok) {
+			addToast({
+				message: `Failed to check out branch. Error ${response.status}: ${response.statusText}`,
+				type: ToastType.Error,
+				dismissible: true
+			});
+			return;
+		}
+
+		// After checking out, call the pull endpoint
+		const pullResponse = await fetch(`${apiAddress}/api/pull/${input}`, {
+			method: 'POST',
+			credentials: 'include'
+		});
+
+		if (!pullResponse.ok) {
+			addToast({
+				message: `Failed to pull latest changes for branch "${input}".`,
+				type: ToastType.Error,
+				dismissible: true
+			});
+			return;
+		} else {
+			addToast({
+				message: `Branch "${input}" checked out and updated successfully.`,
+				type: ToastType.Success,
+				dismissible: true,
+				timeout: 1200
+			});
+		}
```

Comment:
```suggestion
		if (pullResponse.ok) {
			addToast({
				message: `Branch "${input}" checked out and updated successfully.`,
				type: ToastType.Success,
				dismissible: true,
				timeout: 1200
			});
		} else {
			addToast({
				message: `Failed to pull latest changes for branch "${input}".`,
				type: ToastType.Error,
				dismissible: true
			});
			return;
		}
```

I would make it so that this isn't a negative check, IMO it flows more smoothly.